firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/common/util/android.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.vertexai.common.util
+
+import android.media.AudioRecord
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.yield
+
+/**
+ * The minimum buffer size for this instance.
+ *
+ * The same as calling [AudioRecord.getMinBufferSize], except the params are pre-populated.
+ */
+internal val AudioRecord.minBufferSize: Int
+  get() = AudioRecord.getMinBufferSize(sampleRate, channelConfiguration, audioFormat)
+
+/**
+ * Reads from this [AudioRecord] and returns the data in a flow.
+ *
+ * Will emit a zeroed out buffer when this instance is not recording.
+ */
+internal fun AudioRecord.readAsFlow() = flow {
+  val buffer = ByteArray(minBufferSize)
+  val emptyBuffer = ByteArray(minBufferSize)
+
+  while (true) {
+    if (recordingState != AudioRecord.RECORDSTATE_RECORDING) {
+      yield()
```

Comment:
so we are actually not emitting any empty data here? so when the server is talking we are not sending any data?

Reply:
Actually, I think I've found a solution. Will have it out soon.

Reply:
Fixed